src/main.ts

```diff
@@ -138,10 +139,34 @@ function createIdGenerator(): () => Id {
   }
 }
 
-function inferTSConfig(projectPath: string): boolean {
-  // TODO: Move logic from https://github.com/sourcegraph/sourcegraph/pull/30091
-  // into this function.
-  return false;
+/** Returns the list of directory paths under dir which contain a file
+  * with name matchingName.
+*/
+function listDirsRecursive(dir: string, matchingName: string): string[] {
+  let paths: string[] = [];
+  var basenames = fs.readdirSync(dir);
+  for (const basename of basenames) {
+    const path = dir + '/' + basename;
+    const stat = fs.statSync(path);
+    if (stat.isDirectory()) {
+      paths = paths.concat(listDirsRecursive(path, matchingName));

```

Comment:
nit: I typically avoid this `concat` by creating an inner method that pushes to the same shared list

```ts
function listDirsRecursive(dir: string, matchingName: string): string[] {
  const paths: string[] = []
  const loop = (subdir: string) => {
    for (const basename of fs.readdirSync(dir)) { ... }
      if (...) paths.push(...)
  }
  loop(dir)
  return paths
}
```